core/lib/pipeline/object-manipulator.ts

```diff
@@ -5,13 +5,17 @@
 
 import { Clone, CloneAst, DataHandle, DataSink, Descendants, IsPrefix, JsonPath, Mapping, nodes, ParseNode, paths, ReplaceNode, ResolveRelativeNode, SmartPosition, stringify, StringifyAst, ToAst, YAMLNode, ParseToAst } from '@azure-tools/datastore';
 import { From } from 'linq-es2015';
+import { ConfigurationView } from '../autorest-core';
+import { Channel, Message, SourceLocation } from '../message';
 import { IdentitySourceMapping } from '../source-map/merging';
 
 export async function manipulateObject(
   src: DataHandle,
   target: DataSink,
   whereJsonQuery: string,
   transformer: (doc: any, obj: any, path: JsonPath) => any, // transforming to `undefined` results in removal
+  config?: ConfigurationView,
+  extraTransforationContext?: string,
```

Comment:
Small typo, I'd also rename it to this:

```suggestion
  transformationString?: string,
```